packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -54,17 +54,14 @@ export function defaultResolver(
   return realpathSync(result);
 }
 
-export function defaultResolverAsync(
+export async function defaultResolverAsync(
   path: Config.Path,
   options: ResolverOptionsAsync,
 ): Promise<Config.Path> {
   // Yarn 2 adds support to `resolve` automatically so the pnpResolver is only
   // needed for Yarn 1 which implements version 1 of the pnp spec
   if (process.versions.pnp === '1') {
-    // QUESTION: do we need an async version of pnpResolver?
-    // It seems ugly to require a default sync resolver in the async method,
-    // just to deal with this.
-    return Promise.resolve(pnpResolver(path, options));
+    return Promise.resolve(await pnpResolver(path, options));
```

Comment:
The types only work here if we can update the types of `jest-pnp-resolver` to be something like this (I'm no TS expert, there's probably a better way):

```
type JestResolverOptions<T> = {
  basedir: string;
  defaultResolver: T;
  extensions?: Array<string>;
};

type DefaultResolverSync = (request: string, opts: any) => string;
type DefaultResolverAsync = (request: string, opts: any) => Promise<string>;
type DefaultResolver = DefaultResolverSync | DefaultResolverAsync;


type Inferred<T> = T extends JestResolverOptions<infer D> ? D : never;

export default function resolve<D>(request: string, opts: JestResolverOptions<D>): Inferred<D>;
```

I think this should work, though, because in end of the day, jest-pnp-resolver just returns a call to the default resolver (https://github.com/arcanis/jest-pnp-resolver/blob/d7034d2678b2dcbccccc5155b09bb6794508d129/index.js#L48)

Reply:
Seems reasonable to me 👍 You can add a `@ts-ignore` comment in the meantime, and when we're happy with the API here we can send them a PR 🙂

Reply:
I submitted https://github.com/arcanis/jest-pnp-resolver/pull/10.  Even if we don't go this route, that change is technically more accurate than what was there before, and if it's merged/released quickly we won't have to worry about a `@ts-ignore`.